src/main/java/org/opentripplanner/graph_builder/module/ned/ElevationModule.java

```diff
@@ -220,10 +247,68 @@ public void buildGraph(Graph graph, GraphBuilderModuleSummary graphBuilderModule
             "calculate missing elevations"
         );
         log.info(missingElevationsTask.start());
-        assignMissingElevations(graph, edgesWithElevation);
+        assignMissingElevations(graph, edgesWithCalculatedElevations);
         log.info(missingElevationsTask.finish());
     }
 
+    /**
+     * A runnable that contains the relevant info for executing a process edge operation in a particular thread.
+     */
+    private class ProcessEdgeTask implements Runnable {
+        private final StreetWithElevationEdge swee;
+        private final ConcurrentHashMap<Long, Coverage> coveragesForThread;
+
+        public ProcessEdgeTask(StreetWithElevationEdge swee, ConcurrentHashMap<Long, Coverage> coveragesForThread) {
+            this.swee = swee;
+            this.coveragesForThread = coveragesForThread;
+        }
+
+        @Override public void run() {
+            processEdge(swee, getThreadSpecificCoverageInstance());
+        }
+
+        /**
+         * Get the thread-specific Coverage instance to avoid multiple threads waiting for a lock to be released on a
+         * Interpolator2D instance.
+         */
+        private Coverage getThreadSpecificCoverageInstance () {
+            long currentThreadId = Thread.currentThread().getId();
+            Coverage threadSpecificCoverage = coveragesForThread.get(currentThreadId);
+            if (threadSpecificCoverage == null) {
+                // Synchronize the creation of the Thread-specific Coverage instances to avoid potential deadlocks that
+                // could arise from downstream classes that have synchronized methods.
+                synchronized (coveragesForThread) {
+                    threadSpecificCoverage = getCoverage();
```

Comment:
I wouldn't mind a log statement here about initializing a new coverage.

Reply:
It's really not worth it. For the Portland graph, most of these take around 1 millisecond to setup.